### PR TITLE
[e2e] Check errors in kube-state-metrics own metrics

### LINKED ISSUE #1084: Add test to check that no kube-state-metrics own alerts are firing in the e2e test
We have [alerts](https://github.com/kubernetes/kube-state-metrics/blob/master/examples/prometheus-alerting-rules/alerts.yaml) relavant to kube-state-metrics, KubeStateMetricsListErrors and KubeStateMetricsWatchErrors, we need to add a test to our e2e tests that checks the results of the alerting expressions. The expression should be evaluated during the entire run of the e2e tests.

Adding this check in our e2e tests will allow us to make sure that when adding a new metric the resource we are trying to add a list/watch to are not erroring out.

/kind feature

### PR DESCRIPTION
**Pull request by olivierlemasle**
**What this PR does / why we need it**:

This PR adds e2e tests checking that:
- metrics `kube_state_metrics_list_total` and `kube_state_metrics_watch_total` are well present on `/metrics` on telemetry port;
- these metrics do not report any errors.

This PR also moves the "e2e framework" to its own package.

**Which issue(s) this PR fixes**:
Fixes #1084

@lilic In #1084, you mentioned "The expression should be evaluated during the entire run of the e2e tests."
In this PR, I currently check telemetry metrics only once. Would you prefer if I add this check at the end of each other check for example?

### CODE REVIEW

File: tests/e2e/main_test.go [hunk 1]
```diff
@@ -191,3 +193,53 @@ func getLabelsDocumentation() (map[string][]string, error) {
...
 	return documentedMetrics, nil
 }
+
+func TestTelemetryMetrics(t *testing.T) {
+	metricFamilies, err := framework.ParseMetrics(framework.KsmClient.TelemetryMetrics)
+	if err != nil {
+		t.Fatal("Failed to get or decode telemetry metrics", err)
+	}
+
+	// This map'keys are the metrics expected in kube-state-metrics telemetry.
```
Comment (tariq1890) [inline tests/e2e/main_test.go]:
```suggestion
	// This map's keys are the metrics expected in kube-state-metrics' telemetry.
```

File: tests/e2e/main_test.go [hunk 2]
```diff
@@ -191,3 +193,53 @@ func getLabelsDocumentation() (map[string][]string, error) {
...
+	// Its values are a boolean, which is set to true when the metric is found.
```
Comment (tariq1890) [inline tests/e2e/main_test.go]:
```suggestion
	// Its values are boolean, which is set to true when the metric is found.
```

**Comment by tariq1890** [comment]
Thanks @olivierlemasle . PR looks great. Very very minor comments from me.

File: tests/e2e/main_test.go [hunk 3]
```diff
@@ -191,3 +193,53 @@ func getLabelsDocumentation() (map[string][]string, error) {
 	}
 	return documentedMetrics, nil
 }
+
+func TestTelemetryMetrics(t *testing.T) {
```
Comment (lilic) [inline tests/e2e/main_test.go]: I would rename this to something like "TestKubeStateMetricsErrorMetrics", not exactly that but something close.

Comment (lilic) [inline tests/e2e/main_test.go]: When is this test called, are we sure this is called right after everything is run? 🤔

Comment (olivierlemasle) [reply tests/e2e/main_test.go]: @lilic No, it is not!
That's what I explained in the PR's first message, but I was perhaps not clear 😊. I was asking you if would prefer if I called this test after each other test...

Comment (olivierlemasle) [reply tests/e2e/main_test.go]: Ok!

File: tests/e2e/main_test.go (hunk 1 above)
Comment (olivierlemasle) [reply tests/e2e/main_test.go]: Done

File: tests/e2e/main_test.go (hunk 3 above)
Comment (lilic) [reply tests/e2e/main_test.go]: Sorry missed that! At the end would be good, as this metrics only increases we should be able to see any error at the end. 

We seem to run the tests not at the very end, so I guess we need to run the go tests at the end or rethink the bash script? @tariq1890 
https://github.com/kubernetes/kube-state-metrics/blob/master/tests/e2e.sh#L180

Comment (olivierlemasle) [reply tests/e2e/main_test.go]: Yes, we could move go tests execution later. However, as e2e tests are re-written to Go (for example with #950), the bash tests currently executed after the Go ones will disappear.

Comment (lilic) [reply tests/e2e/main_test.go]: yep maybe we should merge that first, good point!

**Comment by brancz** [comment]
Looks good but leaving last call up to @lilic. 

/lgtm
/hold

**Comment by olivierlemasle** [comment]
@lilic Yes, of course!
Done.

**Comment by tariq1890** [comment]
/lgtm

Thanks a lot @olivierlemasle :)

**Comment by lilic** [comment]
Thanks!! 🎉

**Pull request merged**